src/java.desktop/macosx/classes/com/apple/laf/AquaComboBoxUI.java

```diff
@@ -457,7 +457,7 @@ protected LayoutManager createLayoutManager() {
     class AquaComboBoxLayoutManager extends BasicComboBoxUI.ComboBoxLayoutManager {
         protected Rectangle rectangleForCurrentValue() {
             int width = comboBox.getWidth();
-            int height = 22;
+            int height = comboBox.getBorder() == null ? 22 : comboBox.getHeight();
```

Comment:
Before JDK-8054572, comboBox.getHeight()+1 was used , shouldn't we use the same if we are going to fallback to previous way without border

`editorRect.height += 1;`

Also, it seems for TitledBorder the popupmenu is not aligned properly. It should have been at the base of the combobox, just as it is for without border...although I don think it's related to this fix but seems to be a bug nevertheless (in Date/Time macos settings, popupmenu for editable combobox is at the base)

![image](https://user-images.githubusercontent.com/43534309/213630526-9cdc5401-d975-4bc3-9cd8-db3a01057d47.png)

=======
Without border

![image](https://user-images.githubusercontent.com/43534309/213632563-6df07625-22f6-4630-9d3b-511b5ff3cadd.png)

Reply:
Adding 1 to the height would go back to the previous implementation, but it actually causes the comboBox to look more misaligned by 1 extra pixel. I can re-add the +1 to return to the original method if preferred.

<img width="63" alt="Screen Shot 2023-01-20 at 12 54 47 PM" src="https://user-images.githubusercontent.com/96267980/213807706-81de1e18-8922-43f4-8313-72d8332a20b1.png">

I see the popup issue on my end as well. But this seems to be unrelated to the fix as you mentioned. I can create a separate JBS issue for this if needed.

Reply:
OK. I guess that's in Monterey, right? I guess it doesn't cause any difference in BigSur where I tested so I asked.

For popup issue, yes, please raise a JBS..

Reply:
Yes, this is in Monterey. I'll leave the height as is in this change then since it seems better overall.

I'll raise a new JBS issue for the popupmenu on an editable jcombobox.